src/components/PathCrumbs/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { isRegularNode, isRootNode, SchemaNode } from '@stoplight/json-schema-tree';
+import { Box, HStack } from '@stoplight/mosaic';
+import { atom, useAtom } from 'jotai';
+import * as React from 'react';
+
+import { useJSVOptionsContext } from '../../contexts';
+
+export const showPathCrumbsAtom = atom<boolean>(false);
+
+export const pathCrumbsAtom = atom<string[], SchemaNode | readonly string[]>([], (_get, set, node) => {
+  set(pathCrumbsAtom, propertyPathToObjectPath(node as SchemaNode));
+});
+
+export const PathCrumbs = ({ parentCrumbs = [] }: { parentCrumbs?: string[] }) => {
+  const [showPathCrumbs] = useAtom(showPathCrumbsAtom);
+  const [pathCrumbs] = useAtom(pathCrumbsAtom);
+  const { disableCrumbs } = useJSVOptionsContext();
+
+  if (disableCrumbs) {
+    return null;
+  }
+
+  const parentCrumbElems: React.ReactNodeArray = [];
+  parentCrumbs.forEach((crumb, i) => {
+    parentCrumbElems.push(<Box key={i}>{crumb}</Box>);
+  });
+
+  const pathCrumbElems: React.ReactNodeArray = [];
+  pathCrumbs.forEach((crumb, i) => {
+    if (pathCrumbs[i + 1]) {
+      pathCrumbElems.push(<Box key={i}>{crumb}</Box>);
+    } else {
+      pathCrumbElems.push(
+        <Box key={i} color="body" fontWeight="semibold">
+          {crumb}
+        </Box>,
+      );
+    }
+  });
+
+  // only show when have a path, and only when we've scrolled enough such that a portion of the JSV is no longer visible
+  if (!showPathCrumbs || (!parentCrumbElems.length && !pathCrumbElems.length)) {
+    return null;
+  }
+
+  return (
+    <HStack
+      spacing={1}
+      divider={<Box>/</Box>}
+      h="md"
+      // so that the crumbs take up no space in the dom, and thus do not push content down when they appear
+      mt={-8}
+      borderB
+      pos="sticky"
+      top={0}
+      fontFamily="mono"
+      fontSize="sm"
+      lineHeight="none"
+      zIndex={10}
+      bg="canvas-pure"
+      px="px"
+      color="light"
+      alignItems="center"
+    >
+      {parentCrumbElems}
+      {pathCrumbElems.length && <HStack divider={<Box fontWeight="bold">.</Box>}>{pathCrumbElems}</HStack>}
+    </HStack>
+  );
+};
+
+function propertyPathToObjectPath(node: SchemaNode) {
```

Comment:
@marbemac I've changed it a bit.
We could actually use titles if we wanted to.

Reply:
Cool I'll check it out! Re titles that would change the intent of these crumbs. They are meant to allow you to easily understand where a given property sits within the larger object structure - using titles wouldn't achieve that, or maybe I misunderstand?

Reply:
Couple of missed bits - root array should just be `[]` (rather than `undefined[]`):

<img width="813" alt="Screen Shot 2021-11-30 at 10 28 56 AM" src="https://user-images.githubusercontent.com/847542/144076706-d1930003-95de-4a82-a90c-14cb191409cb.png">

Do we want the leading `.` (my original version did not have this). it might confuse people?:

<img width="923" alt="Screen Shot 2021-11-30 at 10 30 05 AM" src="https://user-images.githubusercontent.com/847542/144076906-16a6703a-1689-474a-8d60-96ab8c753303.png">

Reply:
I meant something like `title / title / title`, assuming they are available